crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.move

```diff
@@ -0,0 +1,289 @@
+        assert!(value == 42, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun same_objects_ref_any<Any>(o: &ObjAny<Any>, v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        assert!(o.value == 42, 0);
+        let ObjAny<Any> {id, value: _} = vector::pop_back(&mut v);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+    public entry fun child_access_any<Any>(child: ObjAny<Any>, v: vector<ObjAny<Any>>, _: &mut TxContext) {
+        let ObjAny<Any> {id, value} = child;
+        assert!(value == 42, 0);
+        object::delete(id);
+        let ObjAny<Any> {id, value} = vector::pop_back(&mut v);
+        assert!(value == 42, 0);
+        object::delete(id);
+        vector::destroy_empty(v);
+    }
+
+}
+// "positive" tests start here
+
+
+//# run Test::M::prim_vec_len --sender A --args vector[7,42]
+
+//# run Test::M::mint --sender A --args 42
+
+//# run Test::M::obj_vec_destroy --sender A --args vector[object(107)]
+
+//# run Test::M::mint --sender A --args 42
+
+//# run Test::M::mint_child --sender A --args 42 object(110)
+
+//# run Test::M::child_access --sender A --args object(110) vector[object(112)]
+
+
+// "negative" tests start here
+
+
+//# run Test::M::mint_another --sender A --args 42
+
+//# run Test::M::obj_vec_destroy --sender A --args vector[object(115)]
+
+//# run Test::M::mint_another --sender A --args 42
+
+//# run Test::M::mint --sender A --args 42
+
+//# run Test::M::two_obj_vec_destroy --sender A --args vector[object(118),object(120)]
+
+//# run Test::M::mint_shared --sender A --args 42
+
+//# run Test::M::obj_vec_destroy --sender A --args vector[object(123)]
+
+//# run Test::M::mint --sender A --args 42
+
+//# run Test::M::same_objects --sender A --args object(126) vector[object(126)]
+
+//# run Test::M::mint --sender A --args 42
+
+//# run Test::M::same_objects_ref --sender A --args object(128) vector[object(128)]
+
+
+// "positive" tests start here (for generic vectors)
+
+
+//# run Test::M::mint_any --sender A --type-args Test::M::Any --args 42
+
+//# run Test::M::obj_vec_destroy_any --sender A --type-args Test::M::Any --args vector[object(132)]
+
+//# run Test::M::mint_any --sender A --type-args Test::M::Any --args 42
+
+//# run Test::M::mint_child_any --sender A --type-args Test::M::Any --args 42 object(135)
+
+//# run Test::M::child_access_any --sender A --type-args Test::M::Any --args object(135) vector[object(137)]
+
+
+// "negative" tests start here (for generic vectors)
+
+
+//# run Test::M::mint_another_any --type-args Test::M::Any --sender A --args 42
+
+//# run Test::M::obj_vec_destroy_any --sender A --type-args Test::M::Any --args vector[object(140)]
+
+//# run Test::M::mint_another_any --sender A --type-args Test::M::Any --args 42
+
+//# run Test::M::mint_any --sender A --type-args Test::M::Any --args 42
+
+//# run Test::M::two_obj_vec_destroy_any --sender A --type-args Test::M::Any --args vector[object(143),object(145)]
```

Comment:
Some of these functions seem to be used only once and could probably just be lumped in with the other tests. 

Fun trick for some of these too is you can just `abort` if you don't want to care about the resource rules (cleaningup/deleting the objects)

Reply:
> Some of these functions seem to be used only once and could probably just be lumped in with the other tests.

Agreed, but I personally like functions used for testing a specific feature being defined to test this particular feature only (so that you play with them without worrying about affecting other tests).

> Fun trick for some of these too is you can just `abort` if you don't want to care about the resource rules (cleaningup/deleting the objects)

The way they are, it's easier to use them for by-hand tests, though (e.g., make sure that CLI works for these cases).